### ISSUE TITLE
Streaming error events not handled in anthropic provider

### ISSUE THREAD

**Issue by twillhorn** [vercel/ai#2463]
### Description

I reproduced this by triggering the content filtering policy on Claude 3.5 Sonnet. I am able to do it today with the prompt "generate a really long palindrome".

On the [Vercel AI Playground](https://sdk.vercel.ai/playground), halfway through streaming the response, it will just stop. The playground must be swallowing the error.

In my app I get more details:

```
AI_TypeValidationError: Type validation failed: Value: {"type":"error","error":{"details":null,"type":"invalid_request_error","message":"Output blocked by content filtering policy"}}.
Error message: [
  {
    "code": "invalid_union_discriminator",
    "options": [
      "message_start",
      "content_block_start",
      "content_block_delta",
      "content_block_stop",
      "message_delta",
      "message_stop",
      "ping"
    ],
    "path": [
      "type"
    ],
    "message": "Invalid discriminator value. Expected 'message_start' | 'content_block_start' | 'content_block_delta' | 'content_block_stop' | 'message_delta' | 'message_stop' | 'ping'"
  }
]
digest: "2132174102"
Cause: ZodError: [
  {
    "code": "invalid_union_discriminator",
    "options": [
      "message_start",
      "content_block_start",
      "content_block_delta",
      "content_block_stop",
      "message_delta",
      "message_stop",
      "ping"
    ],
    "path": [
      "type"
    ],
    "message": "Invalid discriminator value. Expected 'message_start' | 'content_block_start' | 'content_block_delta' | 'content_block_stop' | 'message_delta' | 'message_stop' | 'ping'"
  }
]
    at get error (webpack-internal:///(rsc)/./node_modules/.pnpm/zod@3.23.8/node_modules/zod/lib/index.mjs:699:31)
    at eval (eval at safeValidateTypes (webpack-internal:///(rsc)/./node_modules/.pnpm/@ai-sdk+provider-utils@1.0.2_zod@3.23.8/node_modules/@ai-sdk/provider-utils/dist/index.mjs), <anonymous>:1:18)
    at safeValidateTypes (webpack-internal:///(rsc)/./node_modules/.pnpm/@ai-sdk+provider-utils@1.0.2_zod@3.23.8/node_modules/@ai-sdk/provider-utils/dist/index.mjs:234:5)
    at safeParseJSON (webpack-internal:///(rsc)/./node_modules/.pnpm/@ai-sdk+provider-utils@1.0.2_zod@3.23.8/node_modules/@ai-sdk/provider-utils/dist/index.mjs:279:12)
    at Object.transform (webpack-internal:///(rsc)/./node_modules/.pnpm/@ai-sdk+provider-utils@1.0.2_zod@3.23.8/node_modules/@ai-sdk/provider-utils/dist/index.mjs:485:13)
```

From debugging it looks like this is the anthropicMessagesChunkSchema which does not accept "error". Per the Anthropic [API guide](https://docs.anthropic.com/en/api/messages-streaming#error-events), occasional errors should be expected.

**Comment by lgrammel** [comment]
Please update to `@ai-sdk/anthropic@0.0.34`. Duplicate of #2453

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This reports a concrete provider bug: valid Anthropic streaming error events were not accepted or handled, causing stream validation to fail unexpectedly.

### Evidence
- Anthropic streaming can emit a documented `type: "error"` event, but `anthropicMessagesChunkSchema` rejects it and throws `AI_TypeValidationError`.
- The failure causes streams to stop unexpectedly when Claude blocks output through content filtering.
- A maintainer directed the reporter to update to `@ai-sdk/anthropic@0.0.34` and identified it as a duplicate of #2453, indicating the defect was addressed.
